tools/generate_events.py

```diff
@@ -96,7 +96,7 @@
 }
 {%- else %}
 func (e *{{eventName}}) GetClusterId() *strfmt.UUID {
-    {% if event.properties.cluster_id -%}     return &e.ClusterId
+    {% if event.properties.cluster_id -%}     return e.ClusterId
```

Comment:
Could you describe the issues that occurred in the unbound host ? 
If it reduced only to invocation of this method, perhaps it can be changed to
```
    {% if event.properties.cluster_id -%}     if len(e.ClusterId) > 0 {
        return &e.ClusterId
    } else {
      return nil
    }
```
to be nil-safe.

If the problem with with the call to the eventshandler.SendHostEvent(...), then there is a need to change their signatures to support nil values as cluster-id.

Reply:
The specific issue was with the call to the SendHostValidationFallingEvent function (but it may happen to other calls for host events). in previous implementation SendHostValidationFallingEvent received clusterId as strfmt.UUID. in case of the unbound host, clusterId is nil, so calling SendHostValidationFallingEvent (*host.ClusterID) caused segmentation fault. I think it is better simply to store ClusterID as pointer in the host events, since we already have a code that checks if it is nil and calls AddEvent function with appropriate params

Reply:
Okay, in that case @nmagnezi we'll have to use the practice for the infra-env event type which may have the cluster-id optionals in that context